client/gutenberg/editor/utils.js

```diff
@@ -3,29 +3,37 @@
  * External dependencies
  */
 import apiFetch from '@wordpress/api-fetch';
+import wpcomProxyRequest from 'wpcom-proxy-request';
 import { noop } from 'lodash';
 
-/**
- * Internal dependencies
- */
-import debugFactory from 'debug';
-
-const debug = debugFactory( 'calypso:gutenberg' );
-
 export const overrideAPIPaths = siteSlug => {
-	// TODO: no API support for now. We'll also need to handle authorization here.
-	apiFetch.use( () => noop );
-
 	const rootURL = 'https://public-api.wordpress.com/';
 	apiFetch.use( apiFetch.createRootURLMiddleware( rootURL ) );
 
-	// rewrite default API paths to match WP.com equivalents
-	// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
 	apiFetch.use( ( options, next ) => {
-		const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
-
-		debug( 'sending API request to: ', wpcomPath );
-
-		return next( { ...options, path: wpcomPath } );
+		if ( options.path.indexOf( 'wp/v2/sites/' ) >= 0 ) {
```

Comment:
Maybe use `_.startsWith()` (and include the leading slash)?

Reply:
Sure, done in 69233a3251c728c2ae8406ee5147de61c2cd765f.